plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -423,6 +439,11 @@ def set_pre_stop
       end
     end
 
+    def init_containers
+      (JSON.parse(annotations[Kubernetes::Api::Pod::INIT_CONTAINER_KEY] || '[]', symbolize_names: true) +
+        JSON.parse((pod_template.dig(:spec, :initContainers) || '[]'))) || []
```

Comment:
I don't think you'd need to do a `JSON.parse` on `spec.initContainers`.  Only in the annotations do you have a JSON string in the middle of your YAML file.  For `spec.initContainers`, it's normal YAML structure.